### PR TITLE
Fix default HTML list styles

### LINKED ISSUE #1802: [BUG] Fix style issues caused by introduction of default HTML list styling
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug

When a list item contains other HTML elements, e.g., a link, the layout strategy using CSS Grid Layout fails. 

In general, the default list styling can not handle more than 1 direct descendant element of list items.

**Workaround**

A possible workaround is to wrap all content inside each `<li>`, e.g. in a `<div>`, but it is not ideal. It won't solve every case. Vertical alignment will probably also be off.

## Describe how to reproduce the bug

1. Go to e.g. [https://cookbook.kirby.design/#/home/showcase/page](https://cookbook.kirby.design/#/home/showcase/page) or [https://cookbook.kirby.design/#/home/changelog](https://cookbook.kirby.design/#/home/changelog)
1. See error

## Which Kirby version was used?

v4.0.1

## What was the expected behavior?

Proper list styling

## Please complete the following information:
- OS: [e.g. Windows, MacOS, iOS]
- Browser [e.g. Chrome, Safari]
- Browser version [e.g. 22]

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/master/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### PR DESCRIPTION
**Pull request by jkaltoft**
## Which issue does this PR close?

This PR closes #1802 

## What is the new behavior?

This PR replaces what was reverted in #1807.

Kirby applies default styles to HTML lists that differ a bit from user agent styles.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

It may be necessary to adjust list styles where the new defaults are not wanted. Every HTML list will need to be checked individually.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: apps/cookbook/src/app/examples/styling-list-example/styling-list-example.component.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
+<h2>Unordered List Example</h2>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/styling-list-example/styling-list-example.component.html:1]: Consider wrapping each of these examples in a card like we do for https://cookbook.kirby.design/#/home/showcase/badge, to separate them a little more

Comment (RasmusKjeldgaard) [reply apps/cookbook/src/app/examples/styling-list-example/styling-list-example.component.html:1]: We could also consider to make a 2 column layout for these examples, in addition to the above stuff.

File: apps/cookbook/src/app/examples/examples.routes.ts [hunk 2]
```diff
@@ -483,4 +484,8 @@ export const routes: Routes = [
     path: 'link',
     component: LinkExampleComponent,
   },
+  {
+    path: 'styling-list',
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/examples.routes.ts:488]: Should we name this 'Styling HTML Lists' so it is a little more specific when it shows up in the cookbook menu?

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
This seems like a really solid solution. I only have some comments for how it is presented in the cookbook.

Really great work, and a testament to relying on the browser/spec as much as possible!

**Review by RasmusKjeldgaard — COMMENTED** [review]
Refresh my memory - why are we merging this to release/next when there are no breaking changes in this variant of the solution? 

Otherwise, everything looks good!

**Comment by jkaltoft** [comment]
> Refresh my memory - why are we merging this to release/next when there are no breaking changes in this variant of the solution?
> 
> Otherwise, everything looks good!

My memory needs to be refreshed, too 😁. I'll look into it, but I think it still _is_ breaking.

**Comment by jkaltoft** [comment]
> > Refresh my memory - why are we merging this to release/next when there are no breaking changes in this variant of the solution?
> > Otherwise, everything looks good!
> 
> My memory needs to be refreshed, too 😁. I'll look into it, but I think it still _is_ breaking.

Ok, it _is_ breaking. Very, very little, but still, consumer apps should check every occurrence of unordered/ordered lists, e.g., the main menu in relationsbanken.

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**